crates/uv-resolver/src/resolution/display.rs

```diff
@@ -464,16 +460,16 @@ fn combine_extras(graph: &IntermediatePetGraph) -> RequirementsTxtGraph {
         let source = inverse[&source_node.version_id()];
         let target = inverse[&target_node.version_id()];
 
-        // If either the existing marker or new marker is `None`, then the dependency is
-        // included unconditionally, and so the combined marker should be `None`.
+        // If either the existing marker or new marker is `true`, then the dependency is
+        // included unconditionally, and so the combined marker should be `true`.
         if let Some(edge) = next
             .find_edge(source, target)
             .and_then(|edge| next.edge_weight_mut(edge))
         {
-            if let (Some(marker), Some(ref version_marker)) = (edge.as_mut(), weight) {
-                marker.and(version_marker.clone());
+            if edge.is_true() || weight.is_true() {
```

Comment:
This feels a little odd..

Reply:
You could also just OR them and simplify? But this seems simpler.